node/pkg/altpub/alternate_pub.go

```diff
@@ -0,0 +1,534 @@
+
+	for _, ep := range ap.endpoints {
+		ap.logger.Info("Enabling endpoint", zap.String("endpoint", ep.label), zap.String("url", ep.baseUrl), zap.Stringer("delay", ep.delay), zap.Stringer("enabledChains", ep.enabledChains))
+		if ep.delay != 0 {
+			worker := &BatchWorker{ap, ep}
+			common.RunWithScissors(ctx, errC, fmt.Sprintf("alt_pub_batcher_%s", ep.label), worker.batchWorker)
+		}
+	}
+
+	// Wait until shutdown or an error occurs.
+	select {
+	case <-ctx.Done():
+		return ctx.Err()
+	case err := <-errC:
+		return err
+	}
+}
+
+// PublishObservation publishes a signed observation to all the endpoints that care about it. It handles both immediate and delayed publishers.
+func (ap *AlternatePublisher) PublishObservation(emitterChain vaa.ChainID, obs *gossipv1.Observation) {
+	var data []byte
+	for _, ep := range ap.endpoints {
+		if ep.shouldPublish(emitterChain) {
+			if ep.delay == 0 {
+				// We are publishing immediately, build the HTTP request and post it directly to the HTTP worker pool.
+				if data == nil {
+					batch := gossipv1.SignedObservationBatch{
+						Addr:         ap.guardianAddr,
+						Observations: []*gossipv1.Observation{obs},
+					}
+
+					var err error
+					data, err = proto.Marshal((&batch))
+					if err != nil {
+						panic("failed to marshal batch")
+					}
+				}
+
+				req := &HttpRequest{start: time.Now(), ep: ep, url: ep.signedObservationUrl, data: data}
+				select {
+				case ap.httpWorkerChan <- req:
+				default:
+					obsvDropped.WithLabelValues(ep.label).Inc()
+				}
+			} else {
+				// We are batching, post it to the endpoint batching worker.
+				select {
+				case ep.obsvBatchChan <- obs:
+				default:
+					obsvDropped.WithLabelValues(ep.label).Inc()
+				}
+			}
+		}
+	}
+}
+
+// createClient creates the HTTP client using a custom configured transport.
+func (ap *AlternatePublisher) createClient() (*http.Client, error) {
+	defTrans, ok := http.DefaultTransport.(*http.Transport)
+	if !ok {
+		return nil, errors.New("failed to cast default transport for alternate publisher")
+	}
+
+	t := defTrans.Clone()
+	t.MaxIdleConns = MaxIdleConns
+	t.MaxConnsPerHost = MaxConnsPerHost
+	t.MaxIdleConnsPerHost = MaxIdleConnsPerHost
+
+	return &http.Client{
+		Timeout:   HttpClientTimeout,
+		Transport: t,
+	}, nil
+}
+
+// startHttpWorkers starts all of the workers in the HTTP worker pool.
+func (ap *AlternatePublisher) startHttpWorkers(ctx context.Context, client *http.Client, errC chan error, numWorkers int) {
+	for count := range numWorkers {
+		worker := &HttpWorker{ap, client, count}
+		common.RunWithScissors(ctx, errC, "alt_pub_worker", worker.httpWorker)
+	}
+}
+
+// httpWorker is the entrypoint for an HTTP worker. Each httpWorker is responsible for posting an HTTP request and waiting for the response.
+func (w *HttpWorker) httpWorker(ctx context.Context) error {
+	for {
+		select {
+		case <-ctx.Done():
+			return nil
+		case req := <-w.ap.httpWorkerChan:
```

Comment:
Should this part have error handling or a default case?

Reply:
Added error handling. Not sure what a default would do.